src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -447,32 +454,45 @@ export class BrowsingContextImpl {
     this.#cdpTarget.cdpClient.on(
       'Runtime.executionContextCreated',
       (params: Protocol.Runtime.ExecutionContextCreatedEvent) => {
-        if (params.context.auxData.frameId !== this.id) {
-          return;
+        const {auxData, name, uniqueId, id} = params.context;
+
+        let origin = params.context.origin;
+        let sandbox: string | undefined;
+        if (auxData) {
+          // Only these execution contexts are supported for now.
+          switch (auxData.type) {
+            case 'isolated':
+              sandbox = name;
+              // Sandbox should have the same origin as the context itself, but in CDP
+              // it has an empty one.
+              origin = this.#defaultRealm.origin;
+              break;
+            case 'default':
+              break;
+            default:
+              return;
+          }
         }
-        // Only this execution contexts are supported for now.
-        if (!['default', 'isolated'].includes(params.context.auxData.type)) {
-          return;
+        if (['://', ''].includes(origin)) {
```

Comment:
let's extract it in a named helper or keep the comment. Current version does not have an indication why it is so,